### DISCUSSION TITLE
Move some components outside of the `internal-components` folder

### DISCUSSION THREAD

**Discussion by astorije** [vmware-clarity/core#94] [Ideas]
Hi Clarity!

We currently use 2 components from the `internal-components` folder: `CdsInternalCloseButton` and `CdsInternalVisualCheckbox`:

- The former to provide a closable filter panel on our (homegrown) datagrid component:
  <img width="408" alt="Screen Shot 2022-06-08 at 9 20 18 PM" src="https://user-images.githubusercontent.com/113730/172744094-f5139cd6-6ef2-450d-8b44-7f13303632e5.png">
- The latter to provide a presentational-only checkbox in a select that we control otherwise:
  <img width="343" alt="Screen Shot 2022-06-08 at 9 22 07 PM" src="https://user-images.githubusercontent.com/113730/172744243-608a241e-16c6-4857-adb9-07a4eeb15a12.png">

If I remember correctly (@dturcotte could confirm), `CdsInternalVisualCheckbox` was introduced for our team because the regular `CdsCheckbox` suffered from performance issues when loaded in a large list, and interaction issues as the click/hover/etc. is handled by the entire row.
I don't believe Clarity uses it internally. Is there any particular reason this should stay in the `internal-components` folder instead of making it a first-class citizen?

Regarding `CdsInternalCloseButton`, given that v6 now has a whole set of different icon buttons, I think it would make sense to make it a non-internal component as well:

<img width="125" alt="Screen Shot 2022-06-08 at 9 29 08 PM" src="https://user-images.githubusercontent.com/113730/172744897-8a9443d2-7ee7-4de9-a9c9-f6079d7d6005.png">

I can't speak for other internal components as we currently do not use any (thankfully), but I could see how `Popup` might become something to build more complex components from, when needing a popup-like component that is not achievable through existing Clarity components.

What are your thoughts? Thanks again for everything! :)

**Comment by astorije** [comment]
@ashleyryan do you have any thoughts on this?

**Comment by ashleyryan** [comment]
I would prefer to keep the VisualCheckbox internal for now. For us to export it officially, I would want to add a lot more guardrails to it - probably require something like aria-label like icons do, maybe some sort of aria-selected, and I would prefer not to do that at this time.

For the close button, have you looked at the CdsActionButton: https://storybook.core.clarity.design/?path=/story/components-button-action--page

**Reply by astorije** [reply]
Fair enough re: VisualCheckbox, that's understandable.

Re: close button, I mean... sure, we can make it work, but why would we when there is a close button readily available that we know will be consistent with other close buttons within Clarity (e.g. modal)?!
